transcom/mymove
# B 21631 int

## [B-21631](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21631)

## Summary

This PR leverages PDFCPU's LockFormFields API function to lock all SSW PDF Fields after generation is complete. This results in a read-only file as described by the issue, feature, and backlog AC. 

Testing may be added to ensure the locking system works if such tests are feasible, but at this time I do not know whether we could check the ioReader to see if form fields have locked set to true.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the application as a new user, create a move, at least one but preferably multiple PPMs, and request an advance.
2. Log in as a SC. 
3. Approve the advance and orders, view the AOA. Ensure all fields are filled correctly and not editable.
4. Approve the shipment.
5. As a service member, continue the SSW process by uploading documents, weight tickets, etc. and continue the closeout process.
6. Once all details have been added, submit the move again for closeout.
7. Log back in as a SC, complete the closeout.
8. Download the Payment Packet.
9. Ensure that all fields are filled and uneditable.

### Backend


## Screenshots
<img width="553" alt="image" src="https://github.com/user-attachments/assets/7ccb9912-d3cf-483d-becb-1e92c1b2db20">
